### PR TITLE
Adding Deployment example to ROS docs

### PR DESCRIPTION
**Pull request by ruffsl**
Adding deployment example as well as use cases and suggestions.
Also adding dockerfile syntax highlighting.

### CODE REVIEW

**Comment by ruffsl** [comment]
@psftw, I'm adding some more docs to ROS. Similar in style to what is proposed in the PR for the Gazebo docs. Let me know if you have any suggestions.  
Perhaps the deployment example might have to wait, is docker network in v1.7.1 yet?

**Comment by yosifkit** [comment]
Yeah, docker networking plugins are only in experimental builds.

Feeback:
- getting failure on the travis test, related to `markdownfmt` issues.  A diff is printed in the build output.
- no need to run update.sh to also change `README.md` docker-library-bot will do that automatically, so just change content.md
- don't use fenced code blocks, they render horribly on the Docker Hub

File: ros/content.md [hunk 1]
```diff
@@ -19,6 +19,134 @@ You can then build and run the Docker image:
...
+
+### Networks
+
+The ROS runtime "graph" is a peer-to-peer network of processes (potentially distributed across machines) that are loosely coupled using the ROS communication infrastructure. ROS implements several different styles of communication, including synchronous RPC-style communication over services, asynchronous streaming of data over topics, and storage of data on a Parameter Server. To abide by the best practice of [one process per container](https://docs.docker.com/articles/dockerfile_best-practices/), Docker networks can be used to string together several running ROS processes. For further details about [ROS NetworkSetup](http://wiki.ros.org/ROS/NetworkSetup) wik artical, or see the Deployment example below.
+
+## Deployment example
+
+If we want our all ROS nodes to easily talk to each other, we'll can use a virtual network to connect the separate containers. In this short example, we'll create a virtual network, spin up a new container running `roscore` advertised as the `master` service on the new network, then spawn a message publisher and subscriber process as services on the same network.
```
Comment (yosifkit) [inline ros/content.md]: My suggestion is to have this section work without using the docker networking plugins, or have a warning here that lets people know that docker network only works in the experimental docker, or hold back this section until the network plugins are in a release.

Comment (ruffsl) [inline ros/content.md]: Added a note about it. We can remove it after v1.8 release comes around.

**Comment by yosifkit** [comment]
LGTM

File: ros/content.md [hunk 2]
```diff
@@ -19,6 +19,135 @@ You can then build and run the Docker image:
...
+
+## Deployment suggestions
+
+The available tags include supported distros along with a hierarchy tags based off the most common meta-package dependencies, designed to have a small footprint and simple configuration:  
+- `ros-core`: barebone ROS install  
+- `ros-base`: basic tools and libraries (also tagged with distro name with LTS version as `latest`)  
+- `robot`: basic install for robots  
+- `perception`: basic install for perception tasks
+
+The rest of the common meta-packages such as `desktop` and `desktop-full` are hosted on automatic build repos under OSRF's Docker Hub oginsanal profile [here](https://registry.hub.docker.com/u/osrf/ros). These meta-packages include graphical dependencies and hook a host of other large packages such as X11, X server, etc. So in the interest of keep the official images lean and secure, the desktop packages are just be hosted with OSRF's profile.
```
Comment (psftw) [inline ros/content.md]: -oginsanal

File: ros/content.md [hunk 3]
```diff
@@ -19,6 +19,135 @@ You can then build and run the Docker image:
...
+
+	docker run -v "/home/ubuntu/.ros/:/root/.ros/" ros
+
+### Devices
+
+Some application may require device access for acquiring images from connected cameras, control input from human interface device, or GPUS for hardware acceleration. This can be done using the [`--device`](https://docs.docker.com/reference/run/) run argument to mount the device inside the container, providing processes inside hardware access.
+
+### Networks
+
+The ROS runtime "graph" is a peer-to-peer network of processes (potentially distributed across machines) that are loosely coupled using the ROS communication infrastructure. ROS implements several different styles of communication, including synchronous RPC-style communication over services, asynchronous streaming of data over topics, and storage of data on a Parameter Server. To abide by the best practice of [one process per container](https://docs.docker.com/articles/dockerfile_best-practices/), Docker networks can be used to string together several running ROS processes. For further details about [ROS NetworkSetup](http://wiki.ros.org/ROS/NetworkSetup) wik artical, or see the Deployment example below.
```
Comment (psftw) [inline ros/content.md]: wik artical -> wiki article

**Comment by psftw** [comment]
minus the couple of typos I commented on above, very much :+1:

**Comment by ruffsl** [comment]
I'l be making an announcement on our ROS mailing list soon, so I'd to make sure the extra docs are available.
Thanks!

**Comment by yosifkit** [comment]
LGTM, but even once we merge, we'll still be held up by https://github.com/docker/hub-beta-feedback/issues/238.

**Comment by ruffsl** [comment]
Awe shoot. I've been ticketing some issues and giving feedback over there this past week, didn't see that one. Funny to see gazebo third to bottom on that list.

These ones are still bugging the official repos too:
docker/hub-beta-feedback/issues/104
docker/hub-beta-feedback/issues/179

**Pull request merged**